core/dbt/task/init.py

```diff
@@ -48,13 +48,30 @@
 Happy modeling!
 """
 
+class YAMLType(click.ParamType):
+    """Custom YAML type for click parameters and prompts."""
+
+    name = "yaml"
+
+    def convert(
+        self,
+        value: Any,
+        param: Optional[click.Parameter],
+        ctx: Optional[click.Context],
+    ) -> Any:
+        if isinstance(value, str):
+            return yaml.safe_load(value)
```

Comment:
If `value` is an empty string, `yaml.safe_load` will return `None`. I'm pointing it out because we had a previous production bug because of this. Not sure how the `click` library handles `None` values.